inst/tests/tests.Rraw

```diff
@@ -21909,3 +21909,8 @@ DT = suppressMessages(rbindlist(DTl)) # used to crash
 test(2352.1, dim(DT), c(300002L, 1L))
 test(2352.2, DT[[1]], rep(42, nrow(DT)))
 rm(DTn, DTl, DT)
+
+# insert columns by group in freshly unserialized data.tables, #7498
+DT = unserialize(serialize(as.data.table(mtcars), NULL))
+test(2353, DT[,foo:=mean(mpg),by=cyl], as.data.table(mtcars)[,foo:=mean(mpg),by=cyl])
```

Comment:
Add a test that ensure column deletion by `NULL` as value (rather than `NULL` syntactically) is still supported

```r
DT = unserialize(serialize(as.data.table(mtcars), NULL))
null_in_value <- NULL
test(2353a, "cyl" %in% names(DT[, cyl := null_in_value]), FALSE)
DT = unserialize(serialize(as.data.table(mtcars), NULL))
test(2353b,  ncol(DT[, c("cyl", "mpg") := .(null_in_value, null_in_value)]), ncol(mtcars) - 2)
```